pkg/kubelet/config/config.go

```diff
@@ -229,10 +229,18 @@ func (s *podStorage) merge(source string, change interface{}) (adds, updates, de
 				// this is a no-op
 				continue
 			}
-			// this is an add
 			recordFirstSeenTime(ref)
 			pods[name] = ref
-			addPods = append(addPods, ref)
+			// If a pod is not the pending phase and also not in the cache, it
```

Comment:
Or something like "If a pod is not found in the cache, and it's not in the pending phase"